roles/jd-server/src/lib/mempool/mod.rs

```diff
@@ -28,42 +32,57 @@ pub struct TransacrtionWithHash {
 #[derive(Clone, Debug)]
 pub struct JDsMempool {
     pub mempool: Vec<TransacrtionWithHash>,
-    auth: Auth,
+    auth: mini_rpc_client::Auth,
     url: String,
+    receiver: Receiver<String>,
 }
 
 impl JDsMempool {
-    pub fn get_client(&self) -> Option<RpcClient> {
+    pub fn get_client(&self) -> Option<mini_rpc_client::MiniRpcClient> {
         let url = self.url.as_str();
         if url.contains("http") {
-            Some(RpcClient::new(url, self.auth.clone()).unwrap())
+            let client = mini_rpc_client::MiniRpcClient::new(url.to_string(), self.auth.clone());
+            Some(client)
         } else {
             None
         }
     }
-    pub fn new(url: String, username: String, password: String) -> Self {
-        let auth = Auth::UserPass(username, password);
+
+    pub fn get_transaction_list(self_: Arc<Mutex<Self>>) -> Vec<Txid> {
+        let tx_list = self_.safe_lock(|x| x.mempool.clone()).unwrap();
+        let tx_list_: Vec<Txid> = tx_list.iter().map(|n| n.id).collect();
+        tx_list_
+    }
```

Comment:
```suggestion
    }
    
```

Reply:
I don't understand

Reply:
why you don't use `#[cfg(debug_assertions)]`

Reply:
done